.github/workflows/create-azure-self-hosted-runners.yml

```diff
@@ -124,7 +135,32 @@ jobs:
           $ACTIONS_API_URL \
           | jq --raw-output .token)
         echo "::add-mask::$ACTIONS_RUNNER_TOKEN"
-        echo ACTIONS_RUNNER_TOKEN=$ACTIONS_RUNNER_TOKEN >> $GITHUB_ENV
+
+        # The Azure VM type we use has blazing-fast local, temporary storage available as the D:\ drive.
+        # The only downside is that, after dellocation, the contents of this disk (including the Actions Runner),
+        # are destroyed. Let's only use it when we don't immediately deallocate the VM.
+        if [[ "$DEALLOCATE_IMMEDIATELY" == "true" ]]; then
+          ACTIONS_RUNNER_PATH="C:\a"
+        else
+          ACTIONS_RUNNER_PATH="D:\a"
+        fi
+
+        AZURE_ARM_PARAMETERS=$(cat <<-END
+          githubActionsRunnerRegistrationUrl="$ACTIONS_RUNNER_REGISTRATION_URL"
+          githubActionsRunnerToken="$ACTIONS_RUNNER_TOKEN"
+          postDeploymentPsScriptUrl="$POST_DEPLOYMENT_SCRIPT_URL"
+          virtualMachineName="${{ steps.generate-vm-name.outputs.vm_name }}"
+          virtualMachineSize="$AZURE_VM_TYPE"
+          publicIpAddressName1="${{ steps.generate-vm-name.outputs.vm_name }}-ip"
+          adminUsername="${{ secrets.AZURE_VM_USERNAME }}"
+          adminPassword="${{ secrets.AZURE_VM_PASSWORD }}"
+          stopService="$DEALLOCATE_IMMEDIATELY"
+          githubActionsRunnerPath="$ACTIONS_RUNNER_PATH"
+          location="$AZURE_VM_REGION"
+        END
+        )
+
+        echo "AZURE_ARM_PARAMETERS=$AZURE_ARM_PARAMETERS" >> $GITHUB_ENV
```

Comment:
Ah, you need a `<<EOF` trick: see e.g. [this blog post](https://medium.com/@ibraheemabukaff/github-actions-exporting-multi-line-one-line-value-environment-variable-5bb86d01e866).

Reply:
I tested it locally, and `AZURE_ARM_PARAMETERS=$(cat <<-END` actually turns it into a single-line, space-separated string, so then it can be written into the `$GITHUB_ENV` without having to do the `<<EOF` trick you mentioned 🙏🏼

Reply:
Ah, I missed that the `$(...)` aren't enclosed in double-quotes. Excellent!